carla-carma-integration/carla_carma_bridge/src/carla_carma_bridge/carla_to_autoware_vehicle_status

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+
+# Copyright (C) 2021 LEIDOS.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+"""
+Subscribe from CARLA :carla_msgs::CarlaEgoVehicleStatus
+    Topic: /carla/{}/vehicle_status
+
+Subscribe from CARLA :carla_msgs::CarlaEgoVehicleInfo
+    Topic: /carla/{}/vehicle_info
+
+Publish to CARMA :autoware_msgs::VehicleStatus
+    Topic: /hardware_interface/vehicle_status
+
+"""
+import math
+import rospy
+from carla_msgs.msg import CarlaEgoVehicleInfo
+from carla_msgs.msg import CarlaEgoVehicleStatus
+from autoware_msgs.msg import VehicleStatus, Gear
+
+pub = rospy.Publisher('/hardware_interface/vehicle_status', VehicleStatus, queue_size=1)
+vehicle_info = None
+
+
+def vehicle_info_callback(vehicle_info_msg):
+    """
+    callback for vehicle info
+    vehicle_info_msg type:
+        carla_msgs::CarlaEgoVehicleInfo
+    """
+    global vehicle_info
+    vehicle_info = vehicle_info_msg
+
+def vehicle_status_callback(vehicle_status_msg):
+    """
+    callback for vehicle status
+    vehicle_status_msg type:
+        carla_msgs::CarlaEgoVehicleStatus
+    """
+    if vehicle_info is None:
+        return
+    status = VehicleStatus()
+    status.header = vehicle_status_msg.header
+
+    # calculate max steering angle
+    max_steering_angle = math.radians(70)
+
+    # get max steering angle (use smallest non-zero value of all wheels)
+    for wheel in vehicle_info.wheels:
+        if wheel.max_steer_angle:
+            if wheel.max_steer_angle and wheel.max_steer_angle < max_steering_angle:
+                max_steering_angle = wheel.max_steer_angle
+
+    status.angle = vehicle_status_msg.control.steer * math.degrees(max_steering_angle)
```

Comment:
Is `vehicle_status_msg.control.steer` a value in range [-1.0, 1.0]? Do we need to explicitly clamp it into that range or are we safe to assume it'd never be outside of those values?

Also, just for clarity sake, this is wheel angel not steering wheel angle correct?

Reply:
The `vehicle_status_msg.control.steer` is defined in range [-1.0, 1.0] according to the CARLA official [website](https://carla.readthedocs.io/en/0.9.8/ros_msgs/#carlaegovehiclecontrolmsg), so it will not be outside of range.

The script `status.angle = vehicle_status_msg.control.steer * math.degrees(max_steering_angle)` is for calculating wheel angle and set into `VehicleStatus` for CARMA.